packages/@aws-cdk/aws-apigatewayv2/lib/api.ts

```diff
@@ -0,0 +1,186 @@
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as cdk from '@aws-cdk/core';
+import { CfnApi, CfnApiProps,  HttpRouteOptions, IRouteBase, LambdaRouteOptions, Route  } from '../lib';
+
+/**
+ * the HTTP API interface
+ */
+export interface IHttpApi extends cdk.IResource {
+  /**
+   * The ID of this API Gateway HTTP Api.
+   * @attribute
+   */
+  readonly httpApiId: string;
+}
+
+abstract class ApiBase extends cdk.Resource implements IHttpApi {
+  /**
+   * API ID
+   */
+  public abstract readonly httpApiId: string;
+  /**
+   * API URL
+   */
+  public abstract readonly url: string;
+}
+
+/**
+ * API properties
+ */
+export interface HttpApiProps extends cdk.StackProps {
```

Comment:
Drop the extends. This is not an extension of `StackProps`. It should only be used when you're defining your own custom stack.